doc/ref_model/chapters/chapter04.md

```diff
@@ -76,6 +76,9 @@ This section describes a set of explicit NFVI capabilities and performance measu
 | e.nfvi.cap.003 | Total per-instance (ephemeral) storage | GB | Max storage in GB that can be assigned to a single VNFC by NFVI |
 | e.nfvi.cap.004 | # Connection points | number | Max number of connection points that can be assigned to a single VNFC by NFVI |
 | e.nfvi.cap.005 | Total external (persistent) storage | GB | Max storage in GB that can be attached / mounted to VNFC by NFVI |
+| e.nfvi.cap.017 | Total Availability Zones | number | The number of compute availability zones supported by NFVI |
```

Comment:
Total Availability Zones in all of an Operator's Data Centers? What would be the value of knowing that? The number of AZs may be different in each of the OpenStack Regions, for example.

Same comment for storage AZs.

Reply:
I agree with @pgoyal01 comments: it is not clear what we want to achieve here.  In OpenStack AZs are exposed to end users who can select a particular AZ in a particular Region for their VNFC.  This might be an AZ for staging or production or other logical grouping of compute/storage nodes. Hence, this will be dependent  on operator's deployment and LCM architecture.  So the important factors are not the total number (as @pgoyal01 noted) not even number per OS Region, but the purpose  of each AZ, its capability, its capacity and relation to the Region's host aggregates, etc.  What would be the scenario that we could effectively use the knowledge about the number of AZs?

Reply:
> What would be the scenario that we could effectively use the knowledge about the number of AZs?

In particular, what is the desired outcome of any conformance tests that are defined to test this and why are they important in terms of what the CNTT is driving (abstraction, standardisation, etc.)?

Reply:
The desired outcome is a partitioning of compute nodes (and ideally storage resources) that workloads can leverage in a more optimal way.

Without availability zones, workloads are forced to use anti-affinity.  This is fine when there are 2 or even 4 VMs that must not share a failure domain, as anti-affinity would need only 2 or 4 different physical compute hosts. It is not ok when there are 50 or 60 VMs that must not share a failure domain, triggering workload placement on 50 or 60 different compute hosts and, depending on the size of the cluster, exceeding the actual number of compute hosts deployed.

This is a common sort of occurrence with horizontally-scalable, replicated databases.  One wants to ensure that the data is always available by making sure that the data and its replicated copy are not vulnerable to the same failure.  This allows the workload to partition the primary copy of data in one 'az' and the replica in another 'az', but then to not necessarily need to care about whether the DB VMs in the first AZ are sharing underlying compute, since there is no single failure that can take out both the primary copy and its replica.  
The net is that this allows a workload to go from needing 50-60 different compute hosts, down to perhaps 12-20, as the DB VMs in one AZ can now be bin-packed on the same compute hosts.

Having a documented measure, if not a guarantee, of a minimum number of availability zones allows workloads to design in this optimization.  From that standpoint, I see documenting the number, and publishing it, as an important, indirect part of infrastructure abstraction.

This count of AZs is trying to strike a balance between a strict requirement of >=2 AZs per region, which received some negative comments, vs. the need to at least publish and know about the AZ numbers.  

@karinesevilla commented in a different thread that the AZ names are exposed to VNFs.  That's true, and likely should be added as a requirement.

Reply:
@kedmison In OpenStack there is an AZ list command (openstack availability zone list -- compute  (for compute AZs) that lists the available AZs. But please remember AZs in OSTK are host aggregates that are visible to tenants (projects) and where a host can be a member of one AZ only; the latter is different from host aggregates. AZs may not necessarily be failure domains as an AZ in OSTK can be restricted for use of a particular tenant (project). Knowledge of a number of AZs where AZs are restricted for use by certain projects is unlikely to be helpful.

Reply:
@pgoyal01 A count seems to be the wrong way to express this, as there is a lot of differentiation in how you are saying AZs are used.  Can we express this instead as a list of AZs that do separate failure domains? (and thus are appropriate for use in the use case discussed further up the comment thread)?
This would not preclude the presence of other AZs defined for other purposes (like private zones of the data center, or isolating classes of workload)

Reply:
@kedmison Yes -- that would work. It should be changed to a VIM requirement instead of an NFVI capability. Something like, VIM must expose the list of availability zones and their details.

Reply:
@pgoyal01 that's great; thank you.